kernel/src/arch/riscv64/sv39.rs

```diff
@@ -191,20 +211,38 @@ impl ArchitectureMemory for PageTable {
         4096
     }
 
+    // TODO: make this return a Result
     fn map(
         &mut self,
         allocator: &mut mm::PhysicalMemoryManager,
         pa: mm::PAddr,
         va: mm::VAddr,
         perms: mm::Permissions,
     ) {
-        self.map_inner(
-            allocator,
-            PAddr::from_u64(usize::from(pa).try_into().unwrap()),
-            VAddr::from_u64(usize::from(va).try_into().unwrap()),
-            perms,
+        self.map_inner(Some(allocator), pa.into(), va.into(), perms, 2);
+    }
+
+    fn map_noalloc(
+        &mut self,
+        pa: mm::PAddr,
+        va: mm::VAddr,
+        perms: mm::Permissions,
+    ) -> Result<(), Error> {
+        self.map_inner(None, pa.into(), va.into(), perms, 2)?;
+
+        Ok(())
+    }
+
+    fn add_invalid_entry(&mut self, allocator: &mut mm::PhysicalMemoryManager, vaddr: mm::VAddr) {
+        let pte = self.map_inner(
+            Some(allocator),
+            PAddr::from_u64(0x0A0A_0A0A_0A0A_0A0A),
+            vaddr.into(),
+            mm::Permissions::READ,
             2,
-        )
+        );
+        // TODO: remove unwrap when map returns a Result
```

Comment:
You use `map_inner`, which returns a Result. Am I misunderstanding something here?